vpr/src/base/vpr_api.cpp

```diff
@@ -410,6 +413,7 @@ bool vpr_flow(t_vpr_setup& vpr_setup, t_arch& arch) {
             std::cout << "failed placement" << std::endl;
             return false; //Unimplementable
         }
+        print_flat_placement(vpr_setup.FileNameOpts.write_flat_place_file.c_str());
```

Comment:
1. It looks like we unconditionally print a flat placement file (which may be OK; just checking). What is the default filename?
2. I think this print_flat_placement should be pushed down into the vpr_place_flow routine (right by wherever the print_placement call is) since it keeps the high level control routine more lean (easier to understand) and also it's more logical to print all placement files together.

Reply:
I changed it so you have to request it to print a flat placement file after placement; I also added a post-legalizer echo file to print the flat placement (before VPR places orphan clusters).